python/mleap/pyspark/feature.py

```diff
@@ -0,0 +1,38 @@
+from pyspark.ml.util import JavaMLReadable, JavaMLWritable, _jvm
+from pyspark.ml.wrapper import JavaTransformer
+from pyspark.ml.param.shared import HasInputCol, HasOutputCol
+
+
+class StringMap(JavaTransformer, HasInputCol, HasOutputCol, JavaMLReadable, JavaMLWritable):
+    def __init__(self, labels, inputCol=None, outputCol=None, handleInvalid='error', defaultValue=0.0):
+        """
+        __init__(self, labels, inputCol=None, outputCol=None, handleInvalid='error', defaultValue=0.0)
+        labels is a dict {string: double}
+        handleInvalid: how to handle missing labels: 'error' (throw an error), or 'keep' (map to the default value)
+        """
+        assert handleInvalid in ['error', 'keep'], 'Invalid value for handleInvalid: {}'.format(handleInvalid)
+        super(StringMap, self).__init__()
+        labels_scala_map = _jvm() \
+            .scala \
+            .collection \
+            .JavaConverters \
+            .mapAsScalaMapConverter(labels) \
+            .asScala() \
+            .toMap(_jvm().scala.Predef.conforms())
+        handle_invalid_jvm = _jvm().ml.combust.mleap.core.feature.HandleInvalid.__getattr__(
```

Comment:
I think this needs updating to StringMapHandleInvalid.

Reply:
True.

How about this:

> If this should be unit-tested, is there any guidance for how that would be done?

(and I think it should be unit-tested :))

Reply:
I agree about the unit tests, I am not sure we had any for the previous one hot encoder though. Can you take a look at how that could work?

Reply:
WIP pyspark unit testing: https://github.com/combust/mleap/pull/571